File: library/api/src/main/java/org/cru/godtools/api/GlobalActivityAnalyticsApi.java
package org.cru.godtools.api;

import org.ccci.gto.android.common.jsonapi.model.JsonApiObject;
import org.cru.godtools.model.GlobalActivityAnalytics;

import retrofit2.Call;
import retrofit2.http.GET;

public interface GlobalActivityAnalyticsApi {
    String PATH_ANALYTICS_GLOBAL = "analytics/global";

    @GET(PATH_ANALYTICS_GLOBAL)
    Call<JsonApiObject<GlobalActivityAnalytics>> getAnalytics();
}

File: library/global-activity-analytics-manger/src/main/java/org/cru/godtools/global/activity/analytics/manger/api/GlobalActivityAnalyticsApi.kt
package org.cru.godtools.global.activity.analytics.manger.api

import org.ccci.gto.android.common.jsonapi.model.JsonApiObject
import org.cru.godtools.global.activity.analytics.manger.model.GlobalActivityAnalytics
import retrofit2.Call
import retrofit2.http.GET

const val PATH_GLOBAL_ACTIVITY = "/analytics/global"

interface GlobalActivityAnalyticsApi {
    @GET(PATH_GLOBAL_ACTIVITY)
    fun download(): Call<JsonApiObject<GlobalActivityAnalytics>>
}
